ga4gh/backend.py

```diff
@@ -209,6 +214,83 @@ def getReadGroupSets(self):
         """
         return list(self._readGroupSetIdMap.values())
 
+    def getReferenceSets(self):
+        """
+        Returns the list of ReferenceSets in this backend
+        """
+        return list(self._referenceSetIdMap.values())
+
+    def getReference(self, id_):
+        """
+        Returns a reference with the given id_
+        """
+        return self.runGetRequest(self._referenceIdMap, id_)
+
+    def getReferenceSet(self, id_):
+        """
+        Returns a referenceSet with the given id_
+        """
+        return self.runGetRequest(self._referenceSetIdMap, id_)
+
+    def listReferenceBases(self, id_, requestArgs):
+        # parse arguments
+        try:
+            reference = self._referenceIdMap[id_]
+        except KeyError:
+            raise exceptions.ObjectWithIdNotFoundException(id_)
+        start = 0
+        end = datamodel.PysamDatamodelMixin.fastaMax
```

Comment:
Yeah, I'm generally an advocate of "incremental progress with less-than-perfect code, and refactor as it becomes a problem" over "implement clean, very general abstractions before they are necessarily widely needed."  (As has been said in other contexts before, the later is so... java.)  Let's go forward with this PR and file and issue to generalize list request abstractions.

There is still this puzzling CI failure to fix, though.